Add custom tolerations documentation (#724)
okteto/docs: src/content/self-hosted/helm-configuration.mdx

- `persistence.storageClass`: The storage class of the persistence volume attached to every buildkit pod.
- `persistence.size`: The size of the persistence volume attached to every buildkit pod. Defaults to `750Gi`.
- `persistence.cache`: The size (in Mi) of the buildkit cache to store image caches. It should be around 30Gi smaller than `storage.size`. Defaults to 500Gi.

> - `tolerations`: List of tolerations to be added to the buildkit and registry services. This list of tolerations will be applied in conjunction with any tolerations set in [`tolerations.buildPool`](self-hosted/helm-configuration.mdx#tolerations) until deprecation of `tolerations.buildPool`. If not set, the buildkit Pods will inherit the tolerations list set in [`globals.tolerations.okteto`](self-hosted/helm-configuration.mdx#globals).

Reviewer: This should only apply yo `buildkit`, not to the registry

Author: In the chart, the registry-deployment does inherit tolerations from the [`buildkit.tolerations`](https://github.com/okteto/app/blob/05abcb11331d3c3118c3d16bd7769a85deeea019/chart/okteto/templates/registry-deployment.yaml#L77) and was also in the [Custom Tolerations spec](https://www.notion.so/okteto/Custom-Tolerations-in-Chart-61b648a42e7a4d5b89904d6d9823fb2a?pvs=4#e5fc5b24f2a64f5cbbc976601df98232), but let me know if we are no longer treating them as together under the same Build pool, and the `registry` should only inherit from the Okteto tolerations. 

It should be a relatively quick change in the chart and chart tests.

Author: PR for code changes to match documentation changes: https://github.com/okteto/app/pull/8073

Author: Fixed the documentation to reflect that the registry will no longer inherit tolerations from the `buildkit.tolerations` helm chart value.

  priorityClassName: okteto-priority
```

- `tolerations`: Tolerations to be added to Pods that are either under the `okteto` or `dev` categories, as shown below.

>   - `okteto`: A list of tolerations to be added to Pods related to Okteto components (i.e. Okteto api), which run in the `okteto` namespaces. This list of tolerations will be applied in conjunction with any tolerations set in [`tolerations.oktetoPool`](self-hosted/helm-configuration.mdx#tolerations) until deprecation of `tolerations.oktetoPool`.

Reviewer: ```suggestion
  - `okteto`: A list of tolerations to be added to Pods related to Okteto components (i.e. Okteto api), which run in the `okteto` namespace. This list of tolerations will be applied in conjunction with any tolerations set in [`tolerations.oktetoPool`](self-hosted/helm-configuration.mdx#tolerations) until deprecation of `tolerations.oktetoPool`.
```

>   - `dev`: A list of tolerations to be added to Pods related to development workloads (i.e. user application Pods), which run in personal namespaces. This list of tolerations will be applied in conjunction with any tolerations set in [`tolerations.devPool`](self-hosted/helm-configuration.mdx#tolerations) until deprecation of `tolerations.devPool`.

Reviewer: ```suggestion
  - `dev`: A list of tolerations to be added to Pods created by user applications, which run in namespaces managed by Okteto. This list of tolerations will be applied in conjunction with any tolerations set in [`tolerations.devPool`](self-hosted/helm-configuration.mdx#tolerations) until deprecation of `tolerations.devPool`.
```

### tolerations

> Indicates [tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) and [nodeSelector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) for the okteto components.

Reviewer: ```suggestion
Indicates [tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) and [nodeSelector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) for different kind of Pods in the cluster where Okteto is installed.
```
